pkg/daemon/client.go

```diff
@@ -4,25 +4,74 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"io"
+	"log"
 	"net/http"
+	"os"
+	"strconv"
+	"time"
 
 	"connectrpc.com/connect"
 	secretv1 "github.com/codetent/crypta/gen/secret/v1"
 	"github.com/codetent/crypta/gen/secret/v1/secretv1connect"
+	"github.com/hashicorp/go-retryablehttp"
 )
 
 var (
 	ErrSecretNotExists = errors.New("secret does not exist")
 )
 
+const (
+	defaultConnectionTimeout = 5 * time.Second
+	retries                  = 5
+)
+
 type daemonClient struct {
 	client secretv1connect.SecretServiceClient
 }
 
+func calculateRetryTimeout() time.Duration {
+	maxTimeoutStr := os.Getenv("CRYPTA_TIMEOUT")
```

Comment:
Any idea where we could document this env variable?

I also thought about moving the env variables to an `environment.go` file, so we have all of the environment variables in one place. There, they could fill variables that could then be used in other places of the code. I am just not sure how we could handle errors there properly. What do you think?

Reply:
Could make sense. I'm not sure if we probably should use viper in the end for the config management. This would allow to define configs using several ways while giving us the chance to automate the documentation.